### ISSUE TITLE
BoundFieldInitializer for field initializer initializing multiple fields contains only a single field

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn#17813]
Look at this comment for details: https://github.com/dotnet/roslyn/pull/17716#discussion_r105797433

Skipped unit test: Microsoft.CodeAnalysis.VisualBasic.UnitTests.Semantics.IOperationTests.MultipleFieldInitializers

```
Public Sub MultipleFieldInitializers()
            Dim source = <![CDATA[
Class C
	Dim x, y As New Object
End Class
]]>.Value
            Dim compilation = CreateCompilationWithMscorlib(source, options:=TestOptions.ReleaseDll, parseOptions:=TestOptions.Regular)

            Dim tree = compilation.SyntaxTrees.Single()
            Dim node = tree.GetRoot().DescendantNodes().OfType(Of AsNewClauseSyntax)().Single()

            compilation.VerifyOperationTree(node, expectedOperationTree:=<![CDATA[
IFieldInitializer (2 initialized fields) (OperationKind.FieldInitializerAtDeclaration)
  Field_1: C.x As System.Object
  Field_2: C.y As System.Object
  IObjectCreationExpression (Constructor: Sub System.Object..ctor()) (OperationKind.ObjectCreationExpression, Type: System.Object)
]]>.Value)
        End Sub
```

**Comment by mavasani** [comment]
Test fails with following:

```
Message: Expected and actual were different.

Expected: IFieldInitializer (2 initialized fields) (OperationKind.FieldInitializerAtDeclaration)
  Field_1: Field: C.x As System.Object
  Field_2: Field: C.y As System.Object
  IObjectCreationExpression (Constructor: Sub System.Object..ctor()) (OperationKind.ObjectCreationExpression, Type: System.Object)

Actual:   IFieldInitializer (Field: C.x As System.Object) (OperationKind.FieldInitializerAtDeclaration)
  IObjectCreationExpression (Constructor: Sub System.Object..ctor()) (OperationKind.ObjectCreationExpression, Type: System.Object)
```

**Comment by AlekseyTs** [comment]
This is a problem with IOperation implementation.

**Comment by jinujoseph** [comment]
Fred can you check if your changes fixes this as well ?

**Comment by 333fred** [comment]
No, this is an entirely separate API. We're going to need to make small change to IFieldInitializer to support this. However, there's another issue here, in that the BoundFieldInitializer returned here actually does not have both fields, it only has `C.X`.

**Comment by jinujoseph** [comment]
This won't be an issue any more once we merge #21293

**Comment by mavasani** [comment]
Re-activating as we have decided to retain this API - this should be fixed in 15.5
